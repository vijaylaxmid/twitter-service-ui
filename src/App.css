

.tweet-container{
  display: flex;
  height: 100%
}
.tweet-form{
    display: flex;
    flex-direction: column;
    flex: 1;
    height: 100%;
    padding: 20px;
    border-right: 1px solid gray;
}
.tweet-list {
  flex: 2;
  height: 100%;
}

.tweet-app {
  height: 100%;
}
