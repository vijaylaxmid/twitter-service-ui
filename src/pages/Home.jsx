import React from 'react';
import {
  Typography,
} from '@material-ui/core';
import axios from 'axios';
import TweetForm from '../components/TweetForm.jsx';
import TweetList from '../components/TweetList.jsx';
import GetTweetForm from '../components/GetTweetForm.jsx'
export default class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      tweets: []
    }
  }

  getTweet(id) {
    axios.get(`/tweet?id=${id}`)
      .then(res => {
        const tweet = res.data;
        this.setState({ tweets : [tweet] });
      })
  }
  createTweet(text) {
    axios.post(`/tweet`, { text }).then((res) => {
      const tweet = res.data;
      this.setState({ tweets : [tweet, ...this.state.tweets] });
      alert(" created tweet with id:" + tweet.id)
    })
  }

  deleteTweet(id) {
    axios.delete(`/tweet?id=${id}`).then((res) => {
      this.setState({ tweets : this.state.tweets.filter(tweet => tweet.id !== id) });
      alert("delete tweet")
    })
  }

  render () {
    return <div className="tweet-app">
      <Typography component="h3" variant="h3">
        Tweets
      </Typography>
      <div className="tweet-container">
        <div>
          <TweetForm saveTweet={(tweet)=> {
            this.createTweet(tweet)
          } } />
          <GetTweetForm getTweet={(id)=> {
            this.getTweet(id)
          } } />
        </div>
        <TweetList tweets={this.state.tweets}
          deleteTweet={
            (tweetId)=> {
              this.deleteTweet(tweetId)
            }
          }
        />
      </div>
    </div>
  }
}
