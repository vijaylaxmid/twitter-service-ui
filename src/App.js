import React, { Fragment } from 'react';
import {
  CssBaseline,
  withStyles,
} from '@material-ui/core';

import AppHeader from './components/AppHeader';
import Home from './pages/Home';
import './App.css';

const styles = theme => ({
  main: {
    padding: 3 * theme.spacing.unit,
    [theme.breakpoints.down('xs')]: {
      padding: 2 * theme.spacing.unit,
    },
    height: "80vh"
  },
});

const App = ({ classes }) => (
  <Fragment>
    <CssBaseline />
    <AppHeader />
    <main className={classes.main}>
      <Home />
    </main>
  </Fragment>
);

export default withStyles(styles)(App);
