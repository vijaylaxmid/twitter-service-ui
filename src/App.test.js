/**
 author: Vijaylaxmi Nagurkar
*/

import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';
import Home from './pages/Home';


it('App renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});


it('App renders without crashing enzyme', () => {
  shallow(<App />);
});

it('App matches snapshot', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
})

it('Home Page loaded correctly', () => {
  const wrapper = shallow(<Home />);
  expect(wrapper).toMatchSnapshot();
})
