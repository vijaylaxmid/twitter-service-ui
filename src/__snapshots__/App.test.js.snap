// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Home Page loaded correctly 1`] = `ShallowWrapper {}`;
